File: tests/interfile/csharp/constructor_taint/Store.cs
using System;

class Store
{
    private string data;

    public Store(string data)
    {
        this.data = data;
    }

    public void Send()
    {
        // ruleid: test-constructor-taint
        sink(this.data);
    }

    static void sink(string x)
    {
        Console.WriteLine(x);
    }
}

File: tests/interfile/typescript/constructor_taint/Store.ts
export class Store {
    data: string;

    constructor(data: string) {
        this.data = data;
    }

    send(): void {
        // ruleid: test-constructor-taint
        sink(this.data);
    }
}

function sink(x: string): void {
    console.log(x);
}
